frontend/packages/console-shared/src/components/dropdown/dropdown-with-switch/DropdownWithSwitchToggle.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { MenuToggle, Popper } from '@patternfly/react-core';
+
+const DropdownWithSwitchToggle: React.FC<DropdownWithSwitchToggleProps> = ({
+  isDisabled,
+  isFullWidth,
+  isOpen,
+  label,
+  menu,
+  menuRef,
+  onToggle,
+}) => {
+  const toggleRef = React.useRef(null);
+  const containerRef = React.useRef(null);
+  React.useEffect(() => {
+    const handleMenuKeys = (event) => {
```

Comment:
Please also type this event
```suggestion
    const handleMenuKeys = (event: KeyboardEvent) => {
```
Sorry, I sorta mentioned this in my comment to the other handler. You'll also need to bring the `event.target instanceof Node` over as well.